javascript/modules/client/src/client-erc20.ts

```diff
@@ -1,703 +1,52 @@
-     */
-    getSettings: (pluginAddress: string): Promise<IPluginSettings | null> =>
-      this._getSettings(pluginAddress),
-    /**
-     * Returns the details of the token used in a specific plugin instance
-     *
-     * @param {string} pluginAddress
-     * @return {*}  {Promise<Erc20TokenDetails | null>}
-     * @memberof ClientErc20
-     */
-    getToken: (pluginAddress: string): Promise<Erc20TokenDetails | null> =>
-      this._getToken(pluginAddress),
-  };
-
-  //// ACTION BUILDERS
-
-  /** Contains the helpers to encode actions and parameters that can be passed as a serialized buffer on-chain */
-  encoding = {
-    /**
-     * Computes the parameters to be given when creating a proposal that updates the governance configuration
-     *
-     * @param {string} pluginAddress
-     * @param {IPluginSettings} params
-     * @return {*}  {DaoAction}
-     * @memberof ClientErc20
-     */
-    updatePluginSettingsAction: (
-      pluginAddress: string,
-      params: IPluginSettings,
-    ): DaoAction =>
-      this._buildUpdatePluginSettingsAction(pluginAddress, params),
-    /**
-     * Computes the parameters to be given when creating a proposal that mints an amount of ERC-20 tokens to an address
-     *
-     * @param minterAddress
-     * @param params
-     * @returns {*}  {DaoAction}
-     * @memberof ClientErc20
-     */
-    mintTokenAction: (
-      minterAddress: string,
-      params: IMintTokenParams,
-    ): DaoAction => this._buildMintTokenAction(minterAddress, params),
-  };
-  decoding = {
-    /**
-     * Decodes a dao metadata from an encoded update metadata action
-     *
-     * @param {Uint8Array} data
-     * @return {*}  {IPluginSettings}
-     * @memberof ClientErc20
-     */
-    updatePluginSettingsAction: (data: Uint8Array): IPluginSettings =>
-      decodeUpdatePluginSettingsAction(data),
-    /**
-     * Decodes the mint token params from an encoded mint token action
-     *
-     * @param data
-     * @returns {*}  {IMintTokenParams}
-     */
-    mintTokenAction: (data: Uint8Array): IMintTokenParams =>
-      decodeMintTokenAction(data),
-    /**
-     * Returns the decoded function info given the encoded data of an action
-     *
-     * @param {Uint8Array} data
-     * @return {*}  {IInterfaceParams | null}
-     * @memberof ClientErc20
-     */
-    findInterface: (data: Uint8Array): IInterfaceParams | null =>
-      this._findInterfaceParams(data),
-  };
   static encoding = {
     /**
      * Computes the parameters to be given when creating the DAO,
      * so that the plugin is configured
      *
      * @param {IErc20PluginInstall} params
-     * @return {*}  {FactoryInitParams}
+     * @return {*}  {IPluginInstallItem}
      * @memberof ClientErc20
      */
     getPluginInstallItem: (params: IErc20PluginInstall): IPluginInstallItem => {
       return {
-        // id: this._pluginAddress,
         id: PLUGIN_ID,
         data: encodeErc20ActionInit(params),
       };
     },
   };
```

Comment:
Same as before, why not factored out too?

Reply:
see above